tests/main/parallel-install-basic/task.yaml

```diff
@@ -0,0 +1,86 @@
+summary: Checks for parallel installation of a local snap files
+
+prepare: |
+    # ensure we have no snap user data directory yet
+    rm -rf /home/test/snap
+    rm -rf /var/snap/test-snapd-tools /var/snap/test-snapd-tools_foo
+
+execute: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    path="$(make_snap test-snapd-tools)"
+
+    snap set system experimental.parallel-instances=true
+
+    devmode=
+    if [[ "$SPREAD_SYSTEM" = ubuntu-* ]]; then
+        # TODO prallel-install: disable devmode once AppArmor profile updates are in
+        devmode=--devmode
+    fi
+
+    snap install --dangerous --name test-snapd-tools "$path"
+    snap install --dangerous $devmode --name test-snapd-tools_foo "$path"
+
+    su -l -c '! test -d ~/snap/test-snapd-tools' test
+    su -l -c '! test -d ~/snap/test-snapd-tools_foo' test
+
+    su -l -c 'test-snapd-tools_foo.cmd echo foo' test | MATCH foo
+    su -l -c 'test -d ~/snap/test-snapd-tools' test
+    su -l -c 'test -d ~/snap/test-snapd-tools_foo' test
+
+    # instance environment variables are correctly set up
+    su -l -c 'test-snapd-tools_foo.env' test | MATCH 'SNAP_INSTANCE_NAME=test-snapd-tools_foo'
+    su -l -c 'test-snapd-tools_foo.env' test | MATCH 'SNAP_NAME=test-snapd-tools'
+    su -l -c 'test-snapd-tools_foo.env' test | MATCH 'SNAP_INSTANCE_KEY=foo'
+    su -l -c 'test-snapd-tools_foo.env' test | MATCH 'SNAP_COMMON=/var/snap/test-snapd-tools/common'
```

Comment:
Can you please check `$SNAP_DATA` and `$SNAP_USER_DATA` for completeness. This will also catch us making any changes without realizing this.